packages/atomic/src/utils/utils.ts

```diff
@@ -170,3 +170,24 @@ export function isInDocument(element: Node) {
   }
   return false;
 }
+
+export function isPropValuesEqual<ObjectWithProperties extends object>(
+  subject: ObjectWithProperties,
+  target: ObjectWithProperties,
+  propNames: (keyof ObjectWithProperties)[]
+) {
+  return propNames.every((propName) => subject[propName] === target[propName]);
+}
+
+export function getUniqueItemsByProperties<Item extends object>(
+  items: Item[],
+  propNames: (keyof Item)[]
+) {
+  return items.filter(
+    (item, index, self) =>
+      index ===
+      self.findIndex((foundItem) =>
+        isPropValuesEqual(foundItem, item, propNames)
+      )
+  );
```

Comment:
Since this is a util, instead of using a *O*², I think it would be worth improving performance by using a `Set`.

```ts
const existingProps = new Set<string>();
return items.filter(
  (item) => {
    const hash = hashProps(item, propNames);
    if (existingProps.has(hash)) {
      return false;
    }
    existingProps.add(hash);
    return true;
  }
);
```

Reply:
Won't do for now, just seems overkill for the time being.